### PR TITLE
Don't trigger HDAccount DeprecationWarning if Language enum is passed in

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
We made the language that gets passed into the HD Accounts an `Enum`, and deprecated passing the string representation in. However, the DeprecationWarning was getting raised even though an Enum got passed.

### How was it fixed?
Removed extraneous DeprecationWarnings, and changed the return value of `detect_language` because I didn't see it documented anywhere, and therefore I am assuming it was meant to be a private method. Also added a new `list_languages_enum` method. Could stand to come up with a better name for that method, if you have any ideas. Additionally, added a few tests.
### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md) - only added a feature newsfragment because the bugfix portion hadn't been released.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
lgtm

File: eth_account/types.py [hunk 1]
```diff
@@ -35,3 +36,8 @@ class Language(enum.Enum):
     JAPANESE = "japanese"
     KOREAN = "korean"
     SPANISH = "spanish"
+
+    def __lt__(self, other: Any) -> Any:
```
Comment (pacrob) [inline eth_account/types.py:40]: Why is this added?

**Review by pacrob — APPROVED** [review]
Just one question, otherwise lgtm! Just to note, I'm not able to reproduce the `DeprecationWarning` using the `enum` on current `main`. How did you get it to show?

**Review by fselmo — APPROVED** [review]
👍🏼

File: eth_account/types.py (hunk 1 above)
Comment (kclowes) [reply eth_account/types.py:40]: In [this line](https://github.com/ethereum/eth-account/pull/302/files#diff-721e355e9fe8b99722fbb9bfbe182fb03c0929f4601b48ab92ba064d8d14a47cR150), we are sorting the result alphabetically, and the enum didn't know how to do that without adding the `__lt__` method.

**Comment by kclowes** [comment]
> Just to note, I'm not able to reproduce the DeprecationWarning using the enum on current main. How did you get it to show?

In your terminal: 
`$ export PYTHONWARNINGS=always`

and you should see it

**Comment by pacrob** [comment]
> and you should see it

Still doesn't show for me, but no reason to hold this up. It's still an improvement over existing!
